test/sharness/t0115-gateway-dir-listing.sh

```diff
@@ -0,0 +1,146 @@
+#!/usr/bin/env bash
+#
+# Copyright (c) Protocol Labs
+
+test_description="Test directory listing (dir-index-html) on the HTTP gateway"
```

Comment:
This file adds proper end-to-end tests for directory listing on Gateway port that protects us against regressions such as https://github.com/ipfs/dir-index-html/issues/42 by testing each gateway type: 
- path gateway
- subdomain gateway
- dnslink website gateway
    
We check behavior of:
- etag/unicode support
- breadcrumbs ( 
- file name column
- hash column

This level of testing is necessary, because URIs will be different on each gw type, and we really don't want to introduce any regressions around website/directory hosting.